components/match2/wikis/trackmania/match_summary.lua

```diff
@@ -0,0 +1,385 @@
+			:addClass('brkts-popup-comment')
+			:css('white-space','normal')
+			:css('font-size','85%')
+		self.casters = {}
+	end
+)
+
+function Casters:addCaster(caster)
+	if Logic.isNotEmpty(caster) then
+		local nameDisplay = '[[' .. caster.name .. '|' .. caster.displayName .. ']]'
+		if caster.flag then
+			table.insert(self.casters, Flags.Icon(caster['flag']) .. ' ' .. nameDisplay)
+		else
+			table.insert(self.casters, nameDisplay)
+		end
+	end
+	return self
+end
+
+function Casters:create()
+	return self.root
+		:wikitext('Caster' .. (#self.casters > 1 and 's' or '') .. ': ')
+		:wikitext(table.concat(self.casters, #self.casters > 2 and ', ' or ' & '))
+end
+
+-- Custom Header Class
+local Header = Class.new(
+	function(self)
+		self.root = mw.html.create('div')
+		self.root:addClass('brkts-popup-header-dev')
+	end
+)
+
+function Header:leftOpponentTeam(content)
+	self.leftElementAdditional = content
+	return self
+end
+
+function Header:rightOpponentTeam(content)
+	self.rightElementAdditional = content
+	return self
+end
+
+function Header:scoreBoard(content)
+	self.scoreBoard = content
+	return self
+end
+
+function Header:leftOpponent(content)
+	self.leftElement = content
+	return self
+end
+
+function Header:rightOpponent(content)
+	self.rightElement = content
+	return self
+end
+
+function Header:createScoreDisplay(opponent1, opponent2)
+	local function getScore(opponent)
+		local scoreText
+		local isWinner = opponent.placement == 1 or opponent.advances
+		if opponent.placement2 then
+			-- Bracket Reset, show W/L
+			if opponent.placement2 == 1 then
+				isWinner = true
+				scoreText = 'W'
+			else
+				isWinner = false
+				scoreText = 'L'
+			end
+		elseif opponent.extradata and opponent.extradata.additionalScores then
+			-- Match Series (Sets), show the series score
+			scoreText = (opponent.extradata.set1win and 1 or 0)
+					+ (opponent.extradata.set2win and 1 or 0)
+					+ (opponent.extradata.set3win and 1 or 0)
+		else
+			scoreText = OpponentDisplay.InlineScore(opponent)
+		end
+		return OpponentDisplay.BlockScore{
+			isWinner = isWinner,
+			scoreText = scoreText,
+		}
+	end
+
+	return mw.html.create('div')
+		:addClass('brkts-popup-spaced')
+		:node(
+			getScore(opponent1)
+				:css('margin-right', 0)
+		)
+		:node(' : ')
+		:node(getScore(opponent2))
+end
+
+function Header:createScoreBoard(score, bestof, isNotFinished)
+	local scoreBoardNode = mw.html.create('div')
+		:addClass('brkts-popup-spaced')
+
+	if String.isNotEmpty(bestof) and bestof > 0 and isNotFinished then
+		return scoreBoardNode
+			:node(mw.html.create('span')
+				:css('line-height', '1.1')
+				:css('width', '100%')
+				:css('text-align', 'center')
+				:node(score)
+			)
+			:node('<br>')
+			:node(mw.html.create('span')
+				:wikitext('(')
+				:node(Abbreviation.make(
+					'Bo' .. bestof,
+					'Best of ' .. bestof
+				))
+				:wikitext(')')
+			)
+	end
+
+	return scoreBoardNode:node(score)
+end
+
+function Header:soloOpponentTeam(opponent, date)
```

Comment:
is this functionality wanted/needed?
if not can just use the header stuff provided via commons

Reply:
Can't find the header provided via commons, do you have a link or something ? I want to compare to better understand what this functionality does exactly in comparison to the default behaviour

Reply:
basically the difference is that rl has the option for solo opponents to display not only the participant but also the team icon of the player in the header
no other wiki has that
other wikis just display the opponentdisplay plus the (total match) score in the matchsummary headers

Reply:
Oh yeah we don't want that
What is the commons Header ? Can't find it..

Reply:
basically you can kick the entire header class that was created in the rl matchsummary 
then just use the commons one like lots wikis do
the following is taken from R6:
![55C4B4CE-9564-4B87-BDB0-6AC2C2621928](https://user-images.githubusercontent.com/75081997/209722067-2474291a-3655-4352-93d5-5992ff9ddd54.jpeg)